File: Java/Algorithms/Selection Sort/selection_sort.java
public class selection_sort {
    public static void print_array(int arr[], int size) {
        for(int i=0; i<size; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    public static void selectionSort(int arr[], int size) {
        for(int i=0; i<size-1; i++) {
            int minIdx = i;
            for(int j=i+1; j<size; j++) {
                if(arr[j] < arr[minIdx]) {
                    minIdx = j;
                }
            }
            int temp = arr[minIdx];
            arr[minIdx] = arr[i];
            arr[i] = temp;
        }
    }
    public static void main(String[] args) {
        int[] arr = new int[] { 121, 43, 11, 34, 5, 34, 65, 567, 23, 238 };
        int size = arr.length;
        System.out.println("initial array :");
        print_array(arr, size);
        System.out.println("sorted array :");
        selectionSort(arr, size);
        print_array(arr, size);
    }
}

File: C/Algorithms/Selection Sort/selection_sort.c
#include<stdio.h>

void printArray(int arr[], int size) {
    int i;
    for(i=0; i<size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}
void swap(int *x, int *y) {
    int temp = *x;
    *x = *y;
    *y = temp;
}
void selectionSort(int arr[], int size) {
    int i;
    for(i=0; i<size-1; i++) {
        int minIdx = i, j = i+1;
        for(j=i+1; j<size; j++) {
            if(arr[minIdx] > arr[j]) {
                minIdx = j;
            }
        }
        swap(&arr[i], &arr[minIdx]);
    }
}

int main() {
    int arr[10] = { 234, 34, 348, 934, 4, 12, 49, 347, 23, 2 };
    int size = sizeof(arr)/sizeof(arr[0]);
    printf("initial array :\n");
    printArray(arr, size);
    selectionSort(arr, size);
    printf("sorted array :\n");
    printArray(arr, size);

    return 0;
}
